panel/widgets/chat.py

```diff
@@ -0,0 +1,1175 @@
+        """
+        Cleanup the exit stack.
+        """
+        if self._exit_stack is not None:
+            self._exit_stack.close()
+            self._exit_stack = None
+        super()._cleanup()
+
+
+class ChatFeed(CompositeWidget):
+    """
+    A widget to display a chat feed with a header and a chat log.
+
+    Parameters
+    ----------
+    value : List[ChatEntry]
+        The entries to add to the chat feed.
+    header : str | Widget
+        The header of the chat feed. Can be a string, pane, or widget.
+    callback : Callable
+        Callback to execute when a user sends a message or
+        when `respond` is called. The signature must include
+        the previous message value `contents`, the previous `user` name,
+        and the component `instance`.
+    callback_user : str
+        The default user name to use for the entry provided by the callback.
+    callback_avatar : str | BinaryIO
+        The default avatar to use for the entry provided by the callback.
+    placeholder : Any
+        Placeholder to display while the callback is running.
+        If not set, defaults to a LoadingSpinner.
+    placeholder_text : str
+        If placeholder is the default LoadingSpinner,
+        the text to display next to it.
+    placeholder_threshold : int
+        Min duration in seconds of buffering before displaying the placeholder.
+        If 0, the placeholder will be disabled.
+    auto_scroll_limit : int
+        Max pixel distance from the latest object in the Column to
+        activate automatic scrolling upon update. Setting to 0
+        disables auto-scrolling.
+    scroll_button_threshold : int
+        Min pixel distance from the latest object in the Column to
+        display the scroll button. Setting to 0
+        disables the scroll button.
+    view_latest : bool
+        Whether to scroll to the latest object on init. If not
+        enabled the view will be on the first object.
+    card_params : Dict
+        Params to pass to Card, like `header`,
+        `header_background`, `header_color`, etc.
+    entry_params : Dict
+        Params to pass to each ChatEntry, like `reaction_icons`, `timestamp_format`,
+        `show_avatar`, `show_user`, and `show_timestamp`.
+    """
+    value = param.List(item_type=ChatEntry, doc="""
+        The entries to add to the chat feed.""")
+
+    header = param.Parameter(doc="""
+        The header of the chat feed. Can be a string, pane, or widget.""")
+
+    callback = param.Callable(doc="""
+        Callback to execute when a user sends a message or
+        when `respond` is called. The signature must include
+        the previous message value `contents`, the previous `user` name,
+        and the component `instance`.""")
+
+    callback_user = param.String(default="Assistant", doc="""
+        The default user name to use for the entry provided by the callback.""")
+
+    callback_avatar = param.ClassSelector(default="🤖", class_=(str, BinaryIO), doc="""
```

Comment:
`BinaryIO` is Ok. But I would prefer being able to specify a `url` to an image. That is easy and awesome.